test/generate_dot_test.cc

```diff
@@ -0,0 +1,142 @@
+  static constexpr double kAngularTolerance{1e-6};
+  static constexpr double kScaleLength{1.0};
+  //@}
+  static constexpr routing::RoutingConstraints kDefaultRoutingConstraints{};
+  const std::string kTShapeRoadFilePath{std::string(kMalidriveResourcesPath) +
+                                        std::string("/resources/odr/TShapeRoad.xodr")};
+  const api::LaneId kStartLaneId{"0_0_1"};
+  const api::LaneId kEndLaneId{"0_0_1"};
+
+  std::unique_ptr<api::RoadNetwork> road_network_{};
+  std::unique_ptr<routing::Router> router_{};
+  routing::graph::Graph graph_{};
+  std::vector<routing::Route> routes_{};
+
+  void SetUp() override {
+    std::map<std::string, std::string> road_network_configuration;
+    road_network_configuration.emplace(malidrive::builder::params::kRoadGeometryId, "malidrive_rg");
+    road_network_configuration.emplace(malidrive::builder::params::kOpendriveFile, kTShapeRoadFilePath);
+    road_network_configuration.emplace(malidrive::builder::params::kLinearTolerance, std::to_string(kLinearTolerance));
+    road_network_configuration.emplace(malidrive::builder::params::kAngularTolerance,
+                                       std::to_string(kAngularTolerance));
+    road_network_configuration.emplace(malidrive::builder::params::kScaleLength, std::to_string(kScaleLength));
+    road_network_configuration.emplace(malidrive::builder::params::kInertialToBackendFrameTranslation, "{0., 0., 0.}");
+    road_network_ = malidrive::loader::Load<malidrive::builder::RoadNetworkBuilder>(road_network_configuration);
+
+    router_ = std::make_unique<maliput::DistanceRouter>(*road_network_, kLinearTolerance);
+
+    graph_ = routing::graph::BuildGraph(road_network_->road_geometry());
+
+    const api::RoadPosition start(road_network_->road_geometry()->ById().GetLane(kStartLaneId),
+                                  api::LanePosition(1., 0., 0.));
+    const api::RoadPosition end(road_network_->road_geometry()->ById().GetLane(kEndLaneId),
+                                api::LanePosition(10., 0., 0.));
+    routes_ = router_->ComputeRoutes(start, end, kDefaultRoutingConstraints);
+    ASSERT_EQ(1u, routes_.size());
+  }
+};
+
+// Evaluates the graph is correct and the api::Segment containing the routing::Route is marked with the red color.
+TEST_F(TShapeRoadGenerateDotTest, DotGraphWithARouteMarksInRedWhereTheRouteGoesThrough) {
+  const std::string kResult(R"(graph {
+1 -- 5 [ label = "9_0" ];
+5 -- 1 [ label = "8_0" ];
+5 -- 2 [ label = "7_0" ];
+1 -- 2 [ label = "5_0" ];
+4 -- 5 [ label = "2_0" ];
+2 -- 5 [ label = "6_0" ];
+1 -- 2 [ label = "4_0" ];
+2 -- 3 [ label = "1_0" ];
+0 -- 1 [ label = "0_0" color = "red" ];
+}
+)");
+
+  std::stringstream ss;
+  maliput::utility::GenerateDotStream(graph_, routes_.front(), &ss);
+
+  ASSERT_EQ(kResult, ss.str());
+}
+
+// Evaluates the graph serialization is the expected one. There is no prescriptive order in the serialization, thus
+// differences are expected in the order of the edges.
```

Comment:
> There is no prescriptive order in the serialization, thus differences are expected in the order of the edges.

Does the code below handle randomly-ordered edges?

Reply:
[db8afbd](https://github.com/maliput/maliput_integration_tests/pull/85/commits/db8afbd2ce9695f2fe50d579c4132901bcfaf4ea)